pkg/services/searchV2/index.go

```diff
@@ -761,7 +762,7 @@ func (i *searchIndex) updateDashboard(ctx context.Context, orgID int64, index *o
 
 	var folderUID string
 	if dash.folderID == 0 {
-		folderUID = "general"
```

Comment:
@ArturWierzbicki @ryantxu I've looked at how search v2 handles root aka general folder, and made some changes to it in this commit https://github.com/grafana/grafana/pull/60729/commits/f5e940c129702fc9481e44533ac5a79885f91783

Basically, we used to index the general folder, but it's unnecessary and only complicates things. It's enough to set the location of the dashboards to general folder if they have no other parent folder (so that the permissions are resolved correctly), and the frontend takes care of actually displaying the general folder.

But maybe I've missed something. Let me know what you think :)